<h1>Data</h1>

<!-- ========================= -->
<!-- Project Structure Update  -->
<!-- ========================= -->

<h2>Project Structure Update</h2>

<p>
  Since there is no account level information available, everything was shifted to a 
  <strong>profile centric system</strong>.
</p>

<p>
  The project workflow now follows:
  <br />
  <strong>Data → Database → Modelling → Visualisation</strong>
</p>

<hr />

<!-- ========================= -->
<!-- Terminal Setup Steps      -->
<!-- ========================= -->

<h2>Initial Setup Steps (Terminal)</h2>

<pre>
<code>
mkdir streamly-case-study
cd streamly-case-study
mkdir data notebooks backend frontend
</code>
</pre>

<p>
  After creating the folder structure, the CSV files were moved into the 
  <strong>data</strong> directory.
</p>

<hr />

<!-- ========================= -->
<!-- Virtual Environment Setup -->
<!-- ========================= -->

<h2>Virtual Environment Setup</h2>

<p>
  A virtual environment was created and activated.
  <br />
  <strong>Why?</strong> To isolate project dependencies so they do not affect 
  system wide packages or other projects.
</p>

<pre>
<code>
python3 -m venv venv
source venv/bin/activate
pip install pandas numpy jupyter flask
</code>
</pre>

<hr />

<!-- ========================= -->
<!-- Opening the Project in VS Code -->
<!-- ========================= -->

<h2>Opening the Project in VS Code</h2>

<p>
  After setting up the folder structure and virtual environment, open the project directory in VS Code:
</p>

<pre>
<code>
code .
</code>
</pre>

<p>
  Once VS Code opens, select the virtual environment interpreter:
</p>

<ul>
  <li>Open the Command Palette (Ctrl+Shift+P or Cmd+Shift+P).</li>
  <li>Search for <strong>Python: Select Interpreter</strong>.</li>
  <li>Choose the interpreter from the <strong>venv</strong> folder.</li>
</ul>

<p>
  Ensure the Jupyter extension is enabled so notebooks can run inside VS Code.
</p>

In [4]:
# import and load

import pandas as pd
import numpy as np

profiles = pd.read_csv("../data/profiles.csv")
titles = pd.read_csv("../data/titles.csv")

# inspect basic info
print("\n--- PROFILES INFO ---")
profiles.info()
print("\n--- PROFILES SAMPLE ---")
display(profiles.head())

print("\n--- TITLES INFO ---")
titles.info()
print("\n--- TITLES SAMPLE ---")
display(titles.head())


--- PROFILES INFO ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 359 entries, 0 to 358
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   profile_id          359 non-null    int64 
 1   account_id          359 non-null    int64 
 2   profile_name        359 non-null    object
 3   kids_profile        359 non-null    bool  
 4   age_band            359 non-null    object
 5   preferred_language  359 non-null    object
 6   created_at          359 non-null    object
 7   preferences         359 non-null    object
dtypes: bool(1), int64(2), object(5)
memory usage: 20.1+ KB

--- PROFILES SAMPLE ---


,profile_id,account_id,profile_name,kids_profile,age_band,preferred_language,created_at,preferences
0,1,1,Gran,False,35-49,English,2022-10-21,"Horror, Thriller"
1,2,2,Vô,False,25-34,Portuguese,2022-06-12,Horror
2,3,3,Sam,False,35-49,English,2023-02-20,"Horror, Thriller"
3,4,4,Vó,True,13-17,Portuguese,2022-04-18,"Animation, Family"
4,5,4,Pai,False,18-24,Portuguese,2022-04-27,"Comedy, Action, Game-Show"



--- TITLES INFO ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   show_id          1000 non-null   object 
 1   title_name       1000 non-null   object 
 2   category         938 non-null    object 
 3   sub_category     427 non-null    object 
 4   duration         1000 non-null   int64  
 5   age_rating       1000 non-null   object 
 6   type             1000 non-null   object 
 7   year             1000 non-null   int64  
 8   origin_region    994 non-null    object 
 9   language         453 non-null    object 
 10  episode_count    1000 non-null   int64  
 11  is_kids_content  1000 non-null   bool   
 12  imdb_rating      632 non-null    float64
 13  imdb_votes       632 non-null    float64
dtypes: bool(1), float64(2), int64(3), object(8)
memory usage: 102.7+ KB

--- TITLES SAMPLE ---


,show_id,title_name,category,sub_category,duration,age_rating,type,year,origin_region,language,episode_count,is_kids_content,imdb_rating,imdb_votes
0,tt5336140,Gece,Action,Adventure,75,18+,Movie,1989,TR,NaN,1,False,NaN,NaN
1,tt0382083,Hodri Meydan,Adventure,NaN,100,PG,Movie,1962,TR,NaN,1,True,5.9,20.0
2,tt0368651,Civilian Casualties: Fragments from the War on...,Documentary,NaN,60,PG,Movie,2003,US,NaN,1,True,6.1,10.0
3,tt0791389,Salud integral,Documentary,NaN,60,13+,Series,2003,MX,NaN,7,False,NaN,NaN
4,tt29518263,Bob Gurr: Never Say No,Documentary,NaN,75,PG,Movie,2023,CA,NaN,1,True,NaN,NaN


<h2>Initial Data Inspection</h2>

<p>After reviewing the dataset, I performed a basic check of both tables to understand the size and types of values they contain. This helped me see how clean or messy the data is before any modelling or analysis.</p>

<h3>Profiles Table</h3>
<ul>
  <li>Total entries: <strong>359</strong></li>
  <li>The columns contain the following data types:</li>
  <ul>
    <li>1 column with true or false values (boolean)</li>
    <li>2 columns with whole numbers (int64)</li>
    <li>5 columns containing text values (object)</li>
  </ul>
</ul>

<h3>Titles Table</h3>
<ul>
  <li>Total entries: <strong>1000</strong></li>
  <li>The columns contain the following data types:</li>
  <ul>
    <li>1 column with true or false values (boolean)</li>
    <li>2 columns with decimal numbers (float64)</li>
    <li>3 columns with whole numbers (int64)</li>
    <li>8 columns containing text values (object)</li>
  </ul>
</ul>

<h1>Value Distributions / Data Quality</h1>

In [ ]:
print("Profiles age_band value counts:")
print(profiles["age_band"].value_counts())

print("\nProfiles kids_profile value counts:")
print(profiles["kids_profile"].value_counts())

print("\nProfiles preffered_language value counts:")
print(profiles["preferred_language"].value_counts())

print("/nTitles age_rating value counts:")
print(titles["age_rating"].value_counts())

print("\nTitles category value counts (top 20):")
print(titles["category"].value_counts().head(20))

print("\nMissing values per column (profiles):")
print(profiles.isna().sum())

print("\nMissing values per column (titles):")
print(titles.isna().sum())


Missing values per column (titles):
show_id              0
title_name           0
category            62
sub_category       573
duration             0
age_rating           0
type                 0
year                 0
origin_region        6
language           547
episode_count        0
is_kids_content      0
imdb_rating        368
imdb_votes         368
dtype: int64


<h2>Data Quality and Distribution Findings</h2>

<p>While exploring the data, I focused on the distributions and overall quality of the information. Below is a summary of what I found.</p>

<h3>Age Band Distribution</h3>
<p>
The age band <strong>50+</strong> has the highest number of profiles.  
The age band with the lowest number of profiles is <strong>25-34</strong>.  
This shows that most of the users in the dataset are older.
</p>

<h3>Kids vs Non Kids Profiles</h3>
<p>
There are <strong>122 kids profiles</strong> in total.  
Since there are 359 profiles overall, this means <strong>237 profiles are not kids profiles</strong>.
</p>

<h3>Preferred Language</h3>
<p>
Based on the profiles, the most preferred language is <strong>English</strong>.  
The least preferred language is <strong>Zulu</strong>.
</p>

<h3>Age Rating Distribution</h3>
<ul>
  <li><strong>13+</strong> has the highest count with <strong>350</strong> titles.</li>
  <li><strong>PG</strong> follows with <strong>305</strong> titles.</li>
  <li><strong>G (General Audience)</strong> has the lowest count with only <strong>19</strong> titles.</li>
</ul>

<h3>Top 20 Categories</h3>
<ul>
  <li><strong>Drama</strong> appears the most with <strong>231</strong> titles.</li>
  <li><strong>Documentary</strong> is next with <strong>194</strong> titles.</li>
  <li>The least common categories in the top 20 are <strong>Game-Show</strong>, <strong>Sports</strong>, and <strong>Sci-Fi</strong>, each with <strong>5</strong> titles.</li>
</ul>

<h3>Missing Values</h3>

<h4>Profiles Table</h4>
<p>
There are <strong>no missing values</strong> in any of the columns.  
This makes the profiles table clean and easy to work with.
</p>

<h4>Titles Table</h4>
<p>The titles table has several missing values:</p>
<ul>
  <li><strong>Category:</strong> 62 missing</li>
  <li><strong>Sub Category:</strong> 573 missing</li>
  <li><strong>Language:</strong> 547 missing</li>
  <li><strong>IMDB Rating:</strong> 368 missing</li>
  <li><strong>IMDB Votes:</strong> 368 missing</li>
</ul>

<p>This shows that the titles table will need cleaning before any deep modelling or recommendation work.</p>